### DISCUSSION TITLE
computeContactForce in MechanicalContactConstraint cannot be overridden

### DISCUSSION THREAD

**Discussion by npurwar93** [idaholab/moose#25523] [Q&A Modules: Solid mechanics]
I want to have my own implementation of the function 'computeContactForce' in 'MechanicalContactConstraint' class. However, this function is not a virtual function. So, I cannot override it in the derived class. Most of the other functions in this class are virtual. Is there a reason why 'computeContactForce' is not a virtual function? If this could be made a virtual function, it will be helpful in further development of this constraint class.

**Comment by GiudGiud** [comment]
@lindsayad and @bwspenc wrote this function.

I would say make a pull request to make it virtual if you need it

**Reply by lindsayad** [reply]
yea, feel free to make a PR
